src/main/java/org/opentripplanner/middleware/aws/AwsApiGateway.java

```diff
@@ -0,0 +1,104 @@
+package org.opentripplanner.middleware.aws;
+
+import com.amazonaws.services.apigateway.AmazonApiGateway;
+import com.amazonaws.services.apigateway.AmazonApiGatewayClient;
+import com.amazonaws.services.apigateway.AmazonApiGatewayClientBuilder;
+import com.amazonaws.services.apigateway.model.CreateApiKeyRequest;
+import com.amazonaws.services.apigateway.model.CreateApiKeyResult;
+import com.amazonaws.services.apigateway.model.CreateUsagePlanKeyRequest;
+import com.amazonaws.services.apigateway.model.DeleteApiKeyRequest;
+import com.amazonaws.services.apigateway.model.GetUsagePlanRequest;
+import com.amazonaws.services.apigateway.model.GetUsagePlanResult;
+import com.amazonaws.services.apigateway.model.NotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsText;
+
+/**
+ * Manages all interactions with AWS api gateway.
+ */
+public class AwsApiGateway {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AwsApiGateway.class);
+    private static final int SDK_REQUEST_TIMEOUT = 10 * 1000;
+    private static final String AWS_API_GATEWAY_USAGE_PLAN_ID
+        = getConfigPropertyAsText("AWS_API_GATEWAY_USAGE_PLAN_ID");
+
+    /**
+     * Create connection to AWS api gateway.
+     */
+    private static AmazonApiGateway getAmazonApiGateway() {
+        AmazonApiGatewayClientBuilder gatewayBuilder = AmazonApiGatewayClient.builder();
+        AmazonApiGateway gateway = gatewayBuilder.build();
+        return gateway;
+    }
+
+    /**
+     * Request an API key from AWS api gateway and assign it to an existing usage plan.
+     */
+    public static String getApiKeyId(String userId) {
```

Comment:
I think this should probably be called `createApiKey` and return the entire key. We may want to do more with the key in ApiUserController after it's created.

Reply:
Updated.